osi_common.proto

```diff
@@ -1089,3 +1089,15 @@ message KeyValuePair
     //
     optional string value = 2;
 }
+
+//
+// \brief Polygon in 3 dimensions
+//
+// A polygon in 3 dimensions which contains a list of vertices.
+//
+message Polygon3d
+{
+    // A list of vertices
+    //
+    repeated Vector3d vertex = 1;
```

Comment:
Why did you choose Vector3d / cartesian coordinates for the Polygon3d? - I suggest to use Spherical3d instead, because imho it is more common to use angles and ranges to describe the field of view of a sensor.

Reply:
Actually, I wonder why to use Polygon3d anyways, as you could describe the FoV with repeated Spherical3d coordinates already. Did you insert this intermediate step with the surfaces for a specific reason?

In other words: Why do we limit the Virtual detection area to be a composition of surfaces rather then a more flexibel solution? - Sensors like the [Lovox Tele-15](https://www.livoxtech.com/de/tele-15) definetly need more flexibility to be described.

Reply:
I think, Vector3d was used because that's what 3d engines offer for setting up such polygons. Converting this to spherical coordinates for transmission is just another step, I guess. So for our use case of visualizing such area Vector3d is completely fine.

The use case for this PR is just a quite basic visualization of such sensor cones, so we did not want it to become more complicated than it has to be. But I would agree, that this the intention of this detection area is mentioned in the description more specifically.

Reply:
I would say in general the Vector3d approach is more generic.
Using spherical3d defines something which always contains the origin at 0.
So for defining some abstract surface in 3d it will not be possible to use Spherical3d.

The question is  how generic should it be ?

On the other side as Thomas mentioned the Vector3d Format is much better suited towards
rendering applications.

Reply:
edit: I just realized you can also exclude 0 from spherical ...